src/Serializer/ItemNormalizer.php

```diff
@@ -37,11 +39,17 @@ public function normalize($object, $format = null, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
         if (isset($data['id']) && !isset($context['object_to_populate'])) {
+            if (isset($context['allow_update']) && true !== $context['allow_update']) {
```

Comment:
> Have a case where the third-party normalizer is called by our normalizer (very unlikely)

The context is passed down from the serializer itself. Our normalizer does not have to call their normalizer (or vice versa) for there to be a conflict. Merely assigning different semantics to the same key is enough to cause issues.